docs/federation.md

```diff
@@ -0,0 +1,256 @@
+
+Federation is based on the [Federation spec](https://www.apollographql.com/docs/apollo-server/federation/federation-spec/).
+
+A DGS is federation-compatible out of the box with the ability to reference and extend federated types.
+
+!!!tip "There is more federation documentation available"
+    * Read the [Federation Spec](https://www.apollographql.com/docs/apollo-server/federation/federation-spec/).
+    * Check out [Federated Testing](./advanced/federated-testing.md) to learn how to write tests for federated queries.
+
+
+## Federation Example DGS
+
+This is a DGS example that demonstrates how to implement a federated type, and test federated queries.
+The source code in this guide comes from the [Federation example app](https://github.com/Netflix/dgs-federation-example).
+We highly recommend cloning the project and use the IDE while following this guide.
+
+The example project has the following set up:
+
+1. A federated gateway is set up using Apollo's federation gateway libraries.
+2. The [Shows DGS](https://github.com/Netflix/dgs-federation-example/tree/master/shows-dgs) defines and owns the `Show` type.
+3. The [Reviews DGS](https://github.com/Netflix/dgs-federation-example/tree/master/reviews-dgs) adds a `reviews` field to the `Show` type.
+
+
+!!!info
+    If you are completely new to the DGS framework, please take a look at the [DGS Getting Started](./getting-started.md) guide, which also contains an introduction video.
+    The remainder of the guide on this page assumes basic GraphQL and DGS knowledge, and focuses on more advanced use cases.
+
+### Defining the type
+The Shows DGS defines the `Show` type with fields id, title and releaseYear. 
+Note that the `id` field is marked as the key. 
+This indicates to the gateway that the `id` field will be used for identifying the corresponding Show in the Shows DGS and must be specified for federated types.
+```graphql
+type Query {
+  shows(titleFilter: String): [Show]
+}
+
+type Show @key(fields: "id") {
+  id: ID
+  title: String
+  releaseYear: Int
+}
+```
+
+### Extending a Federated Type
+
+To extend a type you redefine the type in your own schema, using directive `@extends` to instruct that it's a type extension.
+`@key` is required to indicate the field that the gateway will use to identify the original `Show` for a query.
+In this case, the key is the `id` field.
+
+```graphql
+type Show @key(fields: "id") @extends {
+  id: ID @external
+  reviews: [Review]
+}
+
+type Review {
+  starRating: Int
+}
+```
+When redefining a type, only the id field, and the fields you're adding need to be listed.
+Other fields, such as `title` for `Show` are provided by the Shows DGS and do not need to be specified unless you are using it in the schema.
```

Comment:
Lower case `show`, since we are referencing the field.

Should we add an alert that reads...
> Don't forget to use the `@external` directive if you define a field that doesn't belong to your DGS but you need to reference it.